lib/active_merchant/billing/gateways/checkout_v2.rb

```diff
@@ -163,16 +160,16 @@ def base_url
       end
 
       def avs_result(action, succeeded, response)
-        if succeeded
-          action == :purchase ? AVSResult.new(code: response["card"]["avsCheck"]) : nil
+        if succeeded && action == :purchase || :authorize
```

Comment:
I am pretty sure you meant `action == :purchased || action == :authorized`; as-is, `:authorized` is truthy, so this condition is always true.

Reply:
ugh, yes! Thanks for catching.